homeassistant/components/seventeentrack/sensor.py

```diff
@@ -151,6 +152,7 @@ async def async_update(self):
                 {
                     ATTR_FRIENDLY_NAME: package.friendly_name,
                     ATTR_INFO_TEXT: package.info_text,
+                    ATTR_TIMESTAMP: package.timestamp,
```

Comment:
Is the time UTC? All times in state attributes must be UTC time.

Reply:
The 17track API returns those times without TZ info unfortunately. It should be local time for the location of that package. I currently don't have an international package delivery to confirm this assumption but for a package that is travelling in Germany, time is in the German timezone.

As this depends on each package (and for international packages also on package's whereabouts when travelling through multiple timezones) I see no easy and reliable way of converting to UTC. Is that a hard requirement? I've seen localized timestamps on other addons so I wasn't aware this is a must.

Reply:
Yes, it's a hard requirement.

Reply:
I just raised a PR over on py17track to convert the timestamp to UTC. This is based on a timezone the user provides as a config param. If no timezone is provided by the user, timestamps are expected to already be in UTC.

**Example:** 17track returns timestamp "2021-03-01 09:30".
- by default py17track will expect this to be a UTC timestamp and simply return full UTC format "2021-03-01 06:30:00"
- if user provided timezone of "Europe/Berlin" py17track will assume this to be a "Europe/Berlin" timestamp and return it in UTC "2021-03-01 08:30" (which is correct for 1st of March as Europe/Berlin is still in CET (+1h) and not yet CEST (+2h)).

17track API only returns origin and destination country, not a timezone. It's impossible for some countries to return a definite timezone (e.g. United States). It can be assumed that users will mostly track national packages, not international. That's why I decided to let the user enter his/her timezone and base timestamps off of that. They will be incorrectly interpreted for package states where the package is travelling through a different timezone. I can't fix that unfortunately as I can't make up data that 17track did not provide.